### PR TITLE
Beman Standard: add more requirements/recommendations + examples

### PR DESCRIPTION
**Pull request by neatudarius**
#43 

Updates:

A. We decided at our sync to add 2 rules for tests:
1. requirement: DIRECTORY.TESTS
2. requirement: FILE.TEST_NAMES

B. I found other missing pieces, which I am very sure we can quickly solve with our Czar of style - @JeffGarland :
1. FILE.NAMES: I found that we don't have a recommendation for `FILE.NAMES`. I think it should be there, as it's an effect of `LIBRARY.NAMES` (also a recommendation)
2. DIRECTORY.EXAMPLES - we want examples to be in a default location (check https://github.com/beman-project/exemplar/tree/main/examples)
3. DIRECTORY.DOCS - same for documentation
4. DIRECTORY.PAPERS - same for in-repo paper projects

C. Add examples/snippets into the standard, where missing.

### CODE REVIEW

File: docs/beman-standard.md [hunk 1]
```diff
@@ -274,3 +277,8 @@ SPDX-License-Identifier: <SPDX License Expression>
 
 **[FILE.COPYRIGHT]** RECOMMENDATION: Source code files should NOT include a
 copyright notice following the SPDX license identifier.
+
+## C++
```
Comment (neatudarius) [inline docs/beman-standard.md:281]: Not a single change in this section, but I think it's better to have inside C++ file conventions at the end, I expect to expand this section in the future, while the other may be stabilized soon.

image

File: docs/beman-standard.md [hunk 2]
```diff
@@ -235,14 +235,17 @@ must either begin with `detail_` or reside within a subdirectory of
...
+directory. If multiple test types are present, subdirectories can be made (e.g., unit tests, performance etc).
 
-**[CPP.NAMESPACE]**: Headers in `include/beman/<short_name>/` should export
-entities in the `beman::<short_name>` namespace.
+## File layout
 
-## File contents
+**[FILE.NAMES]** RECOMMENDATION: Source code and header should use the `snake_case` naming convention (similar to `LIBRARY.NAMES`).
+
+**[FILE.TEST_NAMES]** REQUIREMENT: Test source code files must use the `*.test.cpp` naming convention.
```
Comment (neatudarius) [inline docs/beman-standard.md:248]: Voted at previous sync

File: docs/beman-standard.md [hunk 3]
```diff
@@ -235,14 +235,17 @@ must either begin with `detail_` or reside within a subdirectory of
 `include/beman/<short_name>/` called `detail` or begins with `detail_`.
 
 **[DIRECTORY.SOURCES]** RECOMMENDATION: Sources and headers not part of the
-public interface should reside in `src/`.
+public interface should reside in the top-level `src/` directory, and should use
+the same structure from `include/` - e.g., `src/beman/<short_name>/`. Check `CMAKE.AVOID_PASSTHROUGHS`.
 
-## C++
+**[DIRECTORY.TESTS]** REQUIREMENT: All test files must reside within the top-level `tests/`
```
Comment (neatudarius) [inline docs/beman-standard.md:241]: Voted at previous sync

File: docs/beman-standard.md [hunk 4]
```diff
@@ -235,14 +235,17 @@ must either begin with `detail_` or reside within a subdirectory of
...
+directory. If multiple test types are present, subdirectories can be made (e.g., unit tests, performance etc).
 
-**[CPP.NAMESPACE]**: Headers in `include/beman/<short_name>/` should export
-entities in the `beman::<short_name>` namespace.
+## File layout
 
-## File contents
+**[FILE.NAMES]** RECOMMENDATION: Source code and header should use the `snake_case` naming convention (similar to `LIBRARY.NAMES`).
```
Comment (neatudarius) [inline docs/beman-standard.md:246]: IMO: Missing piece from the standard, which already is used!

File: docs/beman-standard.md [hunk 5]
```diff
@@ -235,14 +235,17 @@ must either begin with `detail_` or reside within a subdirectory of
 `include/beman/<short_name>/` called `detail` or begins with `detail_`.
 
 **[DIRECTORY.SOURCES]** RECOMMENDATION: Sources and headers not part of the
-public interface should reside in `src/`.
+public interface should reside in the top-level `src/` directory, and should use
+the same structure from `include/` - e.g., `src/beman/<short_name>/`. Check `CMAKE.AVOID_PASSTHROUGHS`.
```
Comment (neatudarius) [inline docs/beman-standard.md:239]: Missing few details here

Comment (JeffGarland) [reply docs/beman-standard.md:261]: I'm not following the part about avoid passthru? Unfortunately that quick web search doesn't turn up docs on that flag so I think we'd need to say *why* we want that -- aka what it does for us

Comment (JeffGarland) [reply docs/beman-standard.md:261]: Ah nevermind -- I see what this refers to now.

Comment (neatudarius) [reply docs/beman-standard.md:261]: It's a reference to another guideline in the same doc.

**Review by camio — APPROVED** [review]

**Pull request merged**